### DISCUSSION TITLE
I wonder why editing DAG files via web UI is not supported?

### DISCUSSION THREAD

**Discussion by jay0129** [apache/airflow#18762] [General] [answered]
Hello all, I am new to Airflow and I wonder if there is a reason why editing DAG files via web UI is not supported.

I found an add-on:
https://github.com/andreax79/airflow-code-editor

However, before I am using this feature, I want to know why default Airflow doesn't have this add-on.

**Comment by potiuk** [comment, marked as answer]
It would be a security issue to enable it. Different people have access to write dags and monitor them. Bring able to change code via UI will give the users of the UI much more power. 

Besides Airflow 2 does not even has the physical possibility of editing the files because wit DAG serialization. UI has no access o the DAG folder at all (DAG structure and code are stored in the DB). But the DAGs (and all relevant code if you need common code) are stored in the DAG folder that is not even available for webserver in case of distributed installation.

Also Airflow UI is for monitoring, not for DAF writing. DAGs ar python code and there are plenty of IDEs and tools (for example GitHub UI if you use GitHub) that can be used for that. There is no reason to duplicate the functionality
